### PR TITLE
Adds Choco-Coins

### LINKED ISSUE #5688: allow chocolate coins to be used in vending machines
no one will ever know or suspect them

### PR DESCRIPTION
**Pull request by 9600bauds**
Also: Liberally applies refresh dialog calls to putting shit in vending machines, adds notice spans to a couple of things.

Choco-Coins can be made with one chocolate bar and 5 milk at a Microwave, or purchased from Getmore's for 75 spessbucks.

Since the coin system is a little shit, this makes it so that you can get additional coins with real money.
This does allow the Chef or anyone who breaks into the kitchen to get coins. However, if you try to cook a coin you'll probably lose the race for glubbs since there are two coins laying around for you already. As far as I know, the rest of premium items aren't very balance-altering and are mostly for fun (correct me if I'm wrong), so ultimately I don't think this will change much if at all. This DOES kind of trivialize the payment on the candymachine, but I have never ever seen anyone use it so I don't think it's necessarily a bad thing.

assblast
Sprites by a generous Spriting Anon in the thread.
Currently lacking a "crumpled gold foil" trash item that would be left behind when unwrapping the coin, but I couldn't sprite it worth a shit.

Fixes  #5688

### CODE REVIEW

**Comment by Imagebard** [comment]
This fixes the emag function for the candy machine. If you put coins after its emagged, it shows special messages and dispenses a "strange" candy. You can check it by downloading the files and looking at /obj/item/weapon/reagent_containers/food/snacks/sweet/strange 

>YOU RUINED MUH SEKRITS FUCKING SHITTER

(lol)

**Comment by 9600bauds** [comment]
Uh... That's how it already worked before, I didn't change the functionality

I'm not exactly sure what are you even trying to say

**Comment by Iamgoofball** [comment]
Pfft
:+1:

**Comment by Arthurdentist** [comment]
as the person who added the candy machine I think its fine to make sweets cheaper, there probably ought to be a way to pay with credits because coins just make it a worthless piece of shit that no one ever uses.

**Comment by AutismSpider** [comment]
@Arthurdentist  make it link up with the pda and shit

**Comment by Kurfursten** [comment]
Seems like it makes the mint completely irrelevant. That being said I don't know if the mint is used a whole lot in the first place due to its location.

**Comment by 9600bauds** [comment]
It doesn't make the mint irrelevant in the sense that this takes a few seconds and one food item to create one coin, while the mint lets you shit out an obscene amount effortlessly. What do you even need more than a few coins for, though?...

**Comment by Kurfursten** [comment]
It's my pet project to add useful items to every vendor for coins. Like for instance, the baster at kitchenware and the special pills at medivend.

**Comment by 9600bauds** [comment]
Maybe this could be limited to chocolate coins only. Or add a specific "chococoin" snack in Getmore's instead of the ones in the candy machine. But the mint system is pretty bad IMO: there's only 2 coins and Miners are the only ones who can get more, and they can get hundreds while they're at it. Not like it ever happens,  though.

**Comment by Kurfursten** [comment]
Yeah, it would be nice if the mint were handled differently. Maybe the mint should be moved to a public area (think Recycling) and then people could make iron coins to their heart's content. This has the same problem of making coins way more common, but at least it wouldn't have the weird situation where you're more likely to get a coin from the chef than a miner.

I do like the idea of restricting it to chocolate coins though, that would require the chef to specifically make a chocolate bar, right?

**Comment by 9600bauds** [comment]
Proooobably shouldn't be using the candy machine's product. I like the idea of chococoins in Getmore's costing 60 credits or something.
Maybe the mint could be moved to Recycling and it could be made to allow candy as a sheet? (If this is made so, iron coins should not be worth shit)

**Comment by Kurfursten** [comment]
Yeah, avoiding messing with Goofkitchen in general probably avoids a lot of problems in the first place

**Comment by 9600bauds** [comment]
Requesting someone to make a chocolate reskin of the pomfcoin sprite

**Comment by Intigracy** [comment]
Wonderful.

**Comment by Kurfursten** [comment]
Shiny.

**Comment by 9600bauds** [comment]
after thorough input, several polls, and extensive discussion (all pictured above) i believe this is good to go

**Comment by 9600bauds** [comment]
> .dmi conflicts

I hate you all

**Comment by Arthurdentist** [comment]
merge WHEN

**Comment by 9600bauds** [comment]
ty cluster man of optimization

**Comment by DrLeprikon** [comment]
Them sprites so good, so sweet and nice, must've been done by some professional drawer, very good pee arr overall, thumbsup smiley face

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -3880,3 +3880,32 @@
...
+/obj/item/weapon/reagent_containers/food/snacks/chococoin/attack_self(mob/user)
+	if(wrapped)
+		Unwrap(user)
+	else
+		..()
+
+/obj/item/weapon/reagent_containers/food/snacks/chococoin/proc/Unwrap(mob/user)
+	icon_state = "chococoin_unwrapped"
+	to_chat(user, "<span class='notice'>You remove the golden foil from \the [src].</span>")
+	wrapped = 0
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_containers/food/snacks.dm:3911]: Forgot to update description

The result is predictable

Comment (9600bauds) [inline code/modules/reagents/reagent_containers/food/snacks.dm:3911]: Whoosp

**Pull request merged**